apps/gallery/js/ImageEditor.js

```diff
@@ -1221,8 +1230,51 @@ ImageEditor.prototype.getFullSizeBlob = function(type, done, progress) {
       setTimeout(function() {
         done(blob);
       });
+    }
+
+    function updateExif(blob, metaData, callback) {
+      metaData.Orientation = 1;
+      if (metaData.PixelXDimension) {
+        metaData.PixelXDimension = canvas.width;
+      }
+      if (metaData.PixelYDimension) {
+        metaData.PixelYDimension = canvas.height;
+      }
+      JPEGParser.writeExifMetaData(
+        blob, metaData,
+        (error, modifiedBlob) => {
+          if (error) {
+            console.error('Error' + error);
+          }
+          // Process modified file
+          callback(modifiedBlob);
+        });
+    }
+
+    var originalBlob = this.imageBlob;
+    var originalExif = this.edits.EXIF;
+
+    // Finally, convert the edited image to a blob and pass to the callback.
+    canvas.toBlob(function(blob) {
```

Comment:
I didn't notice this on my first review, but it would be good to move the code in gotEncodedBlob() that frees the canvas memory and put it here.  Since the EXIF editing will be somewhat memory intensive, we should release as much memory as soon as we can.  So I think that the only code that needs to remain in gotEncodedBlob() is the stuff that updates the progress bar to 1.0.

At the point when toBlob() is called the progress bar is at 95% complete. It might be good UX to modify the code so that it is at 90% when toBlob() is called, then update it to 95% before updating the EXIF.  I haven't tried it yet, so I don't know if there is any kind of noticeable pause in the progress bar before the edit finishes or not.